dlpy/attribute_utils.py

```diff
@@ -0,0 +1,763 @@
+            
+            else:
+                attr_helper(conn, model_name, attr_set, attr_key, attr_value[0])
+    else:
+        raise TypeError('Extended table attributes must be one of :\n'
+                        '1. character string;\n'
+                        '2. double precision value/list,\n'
+                        '3. int64 value/list,\n'
+                        '4. int value/list,\n'
+                        '5. binary blob.')
+    
+            
+def attr_helper(conn, model_name, attr_set, attr_key, attr_blob):
+            
+    '''
+    Call action to update individual extended model attribute
+
+    Key/value pair required to specify extended attributes.  Provide
+    correct syntax for calling attribute action.
+
+    Parameters
+    ----------
+    conn : CAS
+        The CAS connection object
+    model_name : string
+        Specifies the name of the deep learning model
+    attr_set : string
+        Name of attribute set to update
+    attr_key : string
+        Key name of attribute
+    attr_blob : double, int64, int, char, or binary blob
+        Representation of attribute
+
+    '''
+            
+    # drop existing attribute
+    rt = conn.retrieve('table.attribute',
+                       _messagelevel = 'error',
+                       name=model_name + '_weights',
+                       attributes=[{"key":attr_key}],
+                       set=attr_set,
+                       task="DROP")
+    
+    # NOTE: ignore errors if attribute or attribute set
+    # doesn't exist
+        
+    # add new attribute
+    rt = conn.retrieve('table.attribute',
+                       _messagelevel = 'error',
+                       name=model_name + '_weights',
+                       attributes=[{"key":attr_key,"value":attr_blob}],
+                       set=attr_set,
+                       task="ADD")
+    if rt.severity > 1:
+        for msg in rt.messages:
+            print(msg)
+        raise DLPyError('Cannot add attribute, there seems to be a problem.')
+
+def export_attr_xml(conn, model_name, file_name):
+
+    '''
+    Create XML version of extended attribute table
+
+    Call action to create XML blob containing model attributes.
+    Write resulting blob to text file.
+
+    Parameters
+    ----------
+    conn : CAS
+        The CAS connection object
+    model_name : string
+        Specifies the name of the deep learning model
+    file_name : string
+        Name of XML file
+
+    '''
+
+    rt = conn.retrieve('table.attribute',
+                       _messagelevel = 'error',
+                       name=model_name + '_weights',
+                       task="EXPORT",
+                       xml="attr")
+    if rt.severity > 1:
+        for msg in rt.messages:
+            print(msg)
+        raise DLPyError('Cannot export model attributes, there seems to be a problem.')
+
+    ascii_text = rt['xmlblob'].decode('utf8')
+    myfile = open(file_name, "w")  
```

Comment:
You might want to use a context manager here: 
```
with open(file_name, "w") as myfile:
    myfile.write(ascii_text)
```

Reply:
I can add this.